xclim/sdba/_processing.py

```diff
@@ -105,6 +105,94 @@ def _adapt_freq(
     return xr.Dataset(data_vars={"pth": pth, "dP0": dP0, "sim_ad": sim_ad})
 
 
+# Works with ranks on possibly multiple dimensions (time, window)
+# ds with dimensions (time, window) as input: same dimensions as output
+# Probably that second part could be done by keeping the explicit map block decorator and calling it
+# appropriately with .func?
+#
+# My issue with incorporating all this in `_adapt_freq` is
+# the [[ if "window" in dim: ]] part above. How does wanting a (window,time) output fit with the idea
+# behind the previous function? If I call adapt_freq.func, should I not expect the output to be
+# of dimension (window,time)?
+
+
+def _adapt_freq_s(
+    ds: xr.Dataset,
+    *,
+    dim: Sequence[str],
+    thresh: float = 0,
+) -> xr.Dataset:
+    r"""
+    Adapt frequency of values under thresh of `sim`, in order to match ref.
```

Comment:
I defined `_adapt_freq_s` which is similar to `_adapt_freq.func`, with 2 changes:
* Rank can work on multiple dimensions (i.e. time & window)
* The `window` dimension is not removed from the output dimensions

Maybe this logic could be incorporated directly `_adapt_freq` with a boolean switch.

Reply:
I feel that reducing the window dimension in `_adapt_freq.func` is not what would be expected when `.func` is invoked? My `_adapt_freq_s` is what I expected `_adapt_freq.func` would do for me.

Reply:
Indeed, it would be cleaner if there was only one `_adapt_freq` func.

You could try removing the `sim = ds.sim.isel(window=(ds.sim.window.size - 1) // 2)` (L77-80) in the original function, as I commented, this was only a performance adjustment.

Reply:
Ok, but that still leaves the lines 
```
        dim = dim[0]

        # Get the percentile rank of each value in sim.
        rank = sim.rank(dim, pct=True)
```
`dim=dim[0]` seems motivated by the fact we preemptively select a window value. I will see what happens when using the multi-dimensional rank